src/app/settings/payment.component.ts

```diff
@@ -87,6 +81,38 @@ export class PaymentComponent implements OnInit {
         if (!this.hidePaypal) {
             window.document.head.appendChild(this.btScript);
         }
+        this.theme = await this.storageService.get<string>(ConstantsService.themeKey);
+        if (this.theme == null) {
+            this.theme = await this.platformUtilsService.getDefaultSystemTheme();
+        }
+        if (this.theme === 'light') {
+            this.color = lightInputColor;
+            this.placeholder = lightInputPlaceholderColor;
+        } else {
+            this.color = darkInputColor;
+            this.placeholder = darkInputPlaceholderColor;
+        }
+        this.StripeElementStyle = {
```

Comment:
I'm OK with this being a class property now that we're changing it during runtime. However, I'd structure it as follows:
* `this.StripeElementStyle` and `this.StripeElementClasses` - move these initial assignments to the constructor. Set the `color` values to `null` in the constructor, so that they can be initialized without needing to access to `storageService`.
* then when you're setting the theme colours in `ngOnInit`, instead of having `this.color` and `this.placeholder`, you can just set the values directly, e.g. `this.StripeElementStyle.base.color = lightInputColor`. (or whatever colour it needs to be)
* this also means you can remove `this.color` and `this.placeholder`.

Different way of achieving the same thing, however it feels more straightforward without having so many variables and properties floating around when they're only used in one location.

Reply:
I've pushed this change as discussed